app/ui-react/syndesis/src/modules/integrations/components/editor/EditorSteps.tsx

```diff
@@ -0,0 +1,97 @@
+import { getMetadataValue } from '@syndesis/api';
+import * as H from '@syndesis/history';
+import {
+  ConnectionCard,
+  ConnectionsGrid,
+  ConnectionsGridCell,
+  ConnectionSkeleton,
+} from '@syndesis/ui';
+import { WithLoader } from '@syndesis/utils';
+import * as React from 'react';
+import { Translation } from 'react-i18next';
+import { ApiError, EntityIcon } from '../../../../shared';
+import { IUIStep } from './interfaces';
+
+export interface IEditorStepsProps {
+  error: boolean;
+  loading: boolean;
+  steps: IUIStep[];
+
+  getEditorStepHref(editorStep: IUIStep): H.LocationDescriptor;
+}
+
+export class EditorSteps extends React.Component<IEditorStepsProps> {
+  public render() {
+    return (
+      <Translation ns={['connections', 'shared']}>
+        {t => (
+          <ConnectionsGrid>
+            <WithLoader
+              error={this.props.error}
+              loading={this.props.loading}
+              loaderChildren={
+                <>
+                  {new Array(5).fill(0).map((_, index) => (
+                    <ConnectionsGridCell key={index}>
+                      <ConnectionSkeleton />
+                    </ConnectionsGridCell>
+                  ))}
+                </>
+              }
+              errorChildren={<ApiError />}
+            >
+              {() => {
+                return this.props.steps.map((s, index) => {
+                  const configurationRequired =
```

Comment:
I feel like this could be refactored out into a helper function, probably a function that could take a `Connection | IUIStep`, like:

```
isConfigurationRequired(thingWithBoard: Connection | IUIStep)
```

That way we could have one block of code in the codebase that makes this determination.  I haven't checked but I imagine we could have this pattern in a couple places at this point.

Reply:
Other option is to have the function just accept the bulletin board, actually I like that better, i.e.:

```
isConfigurationRequired(board: BulletinBoard)
```

Then you in all cases you could just call `configurationRequired = isConfigurationRequired(s.board);` and be done with it :-)

Reply:
BTW I wouldn't apply this refactoring to the integration list if you go this route, that flag shows up under slightly different circumstances.

Reply:
Just did this but need to go back and refactor the other parts of the app that could use this, I can do it today but I think it should be separate from this PR.

Reply:
I think it'd be worthwhile to do so now to be honest while you're in there anyways, no?